src/request.hpp

```diff
@@ -25,12 +25,31 @@
 #include "string_ref.hpp"
 
 #include <stdint.h>
+#include <utility>
+#include <vector>
 
 namespace cass {
 
 class Handler;
 class RequestMessage;
 
+class CustomPayload : public RefCounted<CustomPayload> {
+public:
+  CustomPayload(size_t item_count)
+    : items_(item_count) { }
```

Comment:
If we construct a vector with ``item_count`` then append, I think we're storing ``item_count`` empty buffers ahead of the actual items.